x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/plugin/SqlMediaTypeParserTests.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.plugin;
+
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.MediaType;
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.rest.FakeRestRequest;
+import org.elasticsearch.xpack.sql.action.SqlQueryRequest;
+import org.elasticsearch.xpack.sql.proto.Mode;
+import org.elasticsearch.xpack.sql.proto.RequestInfo;
+
+import java.util.Collections;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.sql.plugin.TextFormat.CSV;
+import static org.elasticsearch.xpack.sql.plugin.TextFormat.PLAIN_TEXT;
+import static org.elasticsearch.xpack.sql.plugin.TextFormat.TSV;
+import static org.elasticsearch.xpack.sql.proto.RequestInfo.CLIENT_IDS;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+
+public class SqlMediaTypeParserTests extends ESTestCase {
+    SqlMediaTypeParser parser = new SqlMediaTypeParser();
+
+    public void testPlainTextDetection() {
+        MediaType text = parser.getMediaType(reqWithAccept("text/plain"), createTestInstance(false, Mode.PLAIN, false));
+        assertThat(text, is(PLAIN_TEXT));
+    }
+
+    public void testCsvDetection() {
+        MediaType text = parser.getMediaType(reqWithAccept("text/csv"), createTestInstance(false, Mode.PLAIN, false));
+        assertThat(text, is(CSV));
+    }
+
+    public void testTsvDetection() {
+        MediaType text = parser.getMediaType(reqWithAccept("text/tab-separated-values"), createTestInstance(false, Mode.PLAIN, false));
+        assertThat(text, is(TSV));
+    }
+
+    public void testParametersParsing() {
```

Comment:
Since the methods only tests media type correct detection - i.e. the other parameters in the `Accept` aren't analysed by `SqlMediaTypeParser`, nor tested here, right? -, I would maybe propose to rename it (`testMediaTypeDetection` or similar?).

(Slightly off-topic, the `header` parameter will need a rework or removal, as per #60271, but that's not the object of this PR).